KerbalStuff/blueprints/mods.py

```diff
@@ -557,12 +564,38 @@ def delete_version(mod_id: int, version_id: str) -> werkzeug.wrappers.Response:
         abort(404)
     if version[0].id == mod.default_version_id:
         abort(400)
+
+    protocol = _cfg('protocol')
+    cdn_domain = _cfg('cdn-domain')
+    if protocol and cdn_domain:
+        global _orig_create_connection
+        _orig_create_connection = connection.create_connection
+        connection.create_connection = create_connection_cdn_purge
```

Comment:
Hmm, is this safe in a massively multithreaded environment like SpaceDock? What if another thread attempts to send a CKAN notification while this monkey patch is active, for example?

Reply:
Do you think adding the below mentioned check would reduce the risk of threading collisions enough? It's unlikely that we will do other requests to `spacedock.info` in other places in the backend code in the future, but not impossible.

Reply:
Probably, yes.

But what if two threads try to run this code at the same time?

1. First thread sets `_orig_create_connection` to the default `create_connection` property
2. Second thread sets `_orig_create_connection` to `create_connection_cdn_purge`, because that's what `connection.create_connection` is thanks to the first thread!
3. We have lost track of the stock `create_connection` and cannot restore it

We need a thread-safe way of monkey patching.

Reply:
Another more complicated risk:

1. First thread finishes its request just as second thread wants to start one
2. Second thread sets `connection.create_connection = create_connection_cdn_purge`
3. First thread sets `connection.create_connection = _orig_create_connection`
4. Second thread attempts `requests.request('PURGE',...` which uses the stock `create_connection` because the first thread just reset it, goes to the wrong host, and fails

Reply:
Latest commit intended to solve this.

- Acquire `_orig_create_connection` just once at the very start, don't overwrite
- Only allow one thread at a time to run the monkey patching block (rest will wait)

Reply:
Good solution :+1: